src/anyio/_backends/_asyncio.py

```diff
@@ -65,7 +66,9 @@ def _cancel_all_tasks(loop):
                 return
 
             for task in to_cancel:
-                task.cancel()
+                # Workaround for async generator shutdown on Python 3.6
+                if not isinstance(get_coro(task), async_generator_athrow):
+                    task.cancel()
```

Comment:
you can `sleep(0)` and then cancel these tasks

Reply:
How does `sleep(0)` help?

Reply:
the task will have called `self._coro.send(None)` after sleep(0) finishes